lib/baseProducer.js

```diff
@@ -134,7 +134,7 @@ BaseProducer.prototype.buildPayloads = function (payloads, topicMetadata) {
       if (message instanceof KeyedMessage) {
         return message;
       }
-      return new Message(0, 0, '', message);
+      return new Message(0, 0, p.key || '', message);
```

Comment:
what happens if the key is number `0`? Would it be better to check if key `!= null` and then use the key otherwise empty string?